### PR TITLE
feat: search catalog

### PR DESCRIPTION
**Pull request by BruceClark406**
Adding searchCatalog functionality.

### CODE REVIEW

**Comment by mtwichel** [comment]
https://developer.squareup.com/reference/square/catalog-api/search-catalog-objects

Is that for this endpoint? If so, I think the method is wrong, and all those query params should be in the request body 😕

File: lib/src/functions_model/search_catalog.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  /// Converts a [SearchCatalogResponse] to a [Map]
+  Map<String, dynamic> toJson() => _$SearchCatalogRequestToJson(this);
+
+  final String? cursor;
+  final List<String>? objectTypes;
+  final bool? includeDeletedObjects;
+  final bool? includeRelatedObjects;
+  final String? beginTime;
+  final CatalogQueryType? query;
```
Comment (mtwichel) [inline lib/src/functions_model/search_catalog.dart:31]:
```suggestion
  final CatalogQuery? query;
```

File: lib/src/square_api_client.dart [hunk 2]
```diff
@@ -60,6 +60,11 @@ abstract class SquareApiClient {
     @Query('catalog_version') int? catalogVersion,
   });
 
+  @GET('/v2/catalog/search')
```
Comment (mtwichel) [inline lib/src/square_api_client.dart:63]:
```suggestion
  @POST('/v2/catalog/search')
```

File: lib/src/square_api_client.dart [hunk 3]
```diff
@@ -60,6 +60,11 @@ abstract class SquareApiClient {
...
+  Future<ListCatalogResponse> searchCatalog({
+    @Body() SearchCatalogRequest? searchCatalogRequest,
```
Comment (mtwichel) [inline lib/src/square_api_client.dart:65]:
```suggestion
    @Body() SearchCatalogRequest searchCatalogRequest,
```

**Review by mtwichel — CHANGES_REQUESTED** [review]
Just a few quick things 💙

File: lib/src/functions_model/search_catalog.dart (hunk 1 above)
Comment (BruceClark406) [reply lib/src/functions_model/search_catalog.dart:31]: Do you want CatalogQueryType renamed to CatalogQuery? 
CatalogQuery is not currently defined.

File: lib/src/square_api_client.dart (hunk 2 above)
Comment (BruceClark406) [reply lib/src/square_api_client.dart:63]: nice catch

File: lib/src/square_api_client.dart (hunk 3 above)
Comment (BruceClark406) [reply lib/src/square_api_client.dart:65]: giddy up

File: lib/src/functions_model/search_catalog.dart (hunk 1 above)
Comment (BruceClark406) [reply lib/src/functions_model/search_catalog.dart:31]: @mtwichel

Comment (BruceClark406) [reply lib/src/functions_model/search_catalog.dart]: Yes, it is now fixed

**Review by mtwichel — APPROVED** [review]
LGTM! 🚀

**Pull request merged**